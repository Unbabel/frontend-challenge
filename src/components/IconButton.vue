<template>
  <button class="icon-button" v-on:click="onClick" v-bind:title="title">
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: 'IconButton',
  props: {
    title: String,
    onClick: Function
  }
}
</script>

<style scoped>
button {
  background: none;
  border: none;
  -webkit-appearance: none; 
}
</style>