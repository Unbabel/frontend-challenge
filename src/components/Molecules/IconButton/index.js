import React from 'react'
import PropTypes from 'prop-types'
import styled, { css, keyframes } from 'styled-components'
import { ifProp, prop } from 'styled-tools'
import Icon from '../../Atoms/Icon/'
import Button from '../../Atoms/Button/'

/** *************************************
 *
 * ICON BUTTON Component
 *
 * Variants:
 *
 ************************************** */

const StyledButton = styled(Button)`
  max-width: ${props => props.hasText ? '100%' : '2.5em'};
  width: ${ifProp('hasText', 'auto', '2.5em')};
  padding: ${ifProp('hasText', '0 0.4375em', 0)};
  flex: 0 0 2.5em;
  box-sizing: border-box;
  border:none;
`

const Text = styled.span`
  padding: 0.4375em;
  }
`

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
`

const StyledIcon = styled(Icon)`
  flex: none;
`

const IconButton = ({ icon, children, ...props }) => {
  const {right, height,
  } = props
  const iconElement = <StyledIcon height={height ? height / 2.5 : undefined} icon={icon} display="none" />
  return (
    <StyledButton hasText={!!children} {...props}>
      <Wrapper>
        {right || iconElement}
        {children
          && <Text className="text" >{children}</Text>
        }
        {right && iconElement}
      </Wrapper>
    </StyledButton>
  )
}


IconButton.propTypes = {
  icon: PropTypes.string.isRequired,
  responsive: PropTypes.bool,
  breakpoint: PropTypes.number,
  right: PropTypes.bool,
  height: PropTypes.number,
  children: PropTypes.node,
}

IconButton.defaultProps = {
  breakpoint: 420,
}

export default IconButton
