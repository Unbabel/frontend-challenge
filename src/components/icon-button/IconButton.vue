<script setup>
	const emit = defineEmits(["onClickCallback"]);

	defineProps({
		icon: {
			type: String,
			required: true,
		},
	});

	const onClickHandler = () => emit("onClickCallback");
</script>

<template>
	<button
		class="icon-button"
		@click="onClickHandler"
		data-test="icon-button"
		aria-label="Icon Button"
	>
		<img :src="icon" alt="button-icon" />
	</button>
</template>

<style scoped>
	.icon-button {
		background-color: transparent;
		border: none;
	}

	.icon-button:hover {
		cursor: pointer;
	}
</style>
