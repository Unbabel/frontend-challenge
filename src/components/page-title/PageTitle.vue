<script setup>
defineProps({
  text: {
    type: String,
    required: true,
    default: "",
  },
});
</script>

<template>
  <span class="page-title" data-test="page-title">
    {{ text }}
  </span>
</template>

<style scoped>
.page-title {
  font-size: 1.125rem;
  font-weight: 500;
  color: var(--palette-dark);
}
</style>
