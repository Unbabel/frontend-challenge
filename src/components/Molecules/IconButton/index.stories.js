
import React from 'react'
import { storiesOf } from '@storybook/react'
import IconButton from '.'

storiesOf('Molecules/IconButton', module)
  .add('default', () => (
    <IconButton icon="person" transparent />
  ))
  .add('Person', () => (
    <IconButton icon="person" transparent />
  ))
  .add('Upload', () => (
    <IconButton icon="upload" transparent />
  ))
  .add('Delete', () => (
    <IconButton icon="delete" transparent />
  ))
  .add('Fetch Document', () => (
    <IconButton icon="fetch-document" transparent />
  ))
  .add('Add Row', () => (
    <IconButton icon="add-row" transparent />
  ))
  .add('transparent', () => (
    <IconButton icon="person" transparent>Hello</IconButton>
  ))
  .add('with icon on right', () => (
    <IconButton icon="person" right>Hello</IconButton>
  ))

  .add('without text', () => (
    <IconButton icon="person" />
  ))

  .add('height', () => (
    <IconButton icon="person" height={100}>Hello</IconButton>
  ))
