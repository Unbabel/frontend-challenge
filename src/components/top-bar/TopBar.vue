<script setup>
	// Components
	import PageTitle from "../page-title/PageTitle.vue";
	import IconButton from "../icon-button/IconButton.vue";
</script>

<template>
	<div class="top-bar-container">
		<div class="top-bar-inner-container flex-center flex-center-row">
			<PageTitle text="Transcriptions" />
			<div
				class="top-bar-actions-container flex-center flex-center-row"
				data-test="top-bar-actions"
			>
				<IconButton
					v-for="action in $route.meta?.actions"
					@onClickCallback="action.callback"
					:icon="action.icon"
					:key="action.id"
				/>
			</div>
		</div>
	</div>
</template>

<style scoped>
	.top-bar-container {
		width: 100%;
		height: 4.3rem;
		background-color: var(--palette-white);
		box-shadow: 0 1px 3px 0 var(--palette-transp-black);
	}

	.top-bar-inner-container {
		height: 100%;
		width: 55%;
		margin: 0 auto;
		justify-content: space-between;
	}

	.top-bar-actions-container {
		column-gap: 0.5rem;
	}

	@media (max-width: 768px) {
		.top-bar-inner-container {
			width: 90%;
		}
	}
</style>
