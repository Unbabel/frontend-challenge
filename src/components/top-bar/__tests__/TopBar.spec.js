import Vuex from "vuex";
import { describe, it, expect, beforeEach, vi } from "vitest";
import { mount, createLocalVue } from "@vue/test-utils";

// Components
import TopBar from "../TopBar.vue";
import PageTitle from "../../page-title/PageTitle.vue";
import IconButton from "../../icon-button/IconButton.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

const $route = {
	path: "/",
	name: "transcriptions",
	component: null,
	meta: {
		actions: {
			icon: "/path/to/icon.svg",
			callback: vi.fn(),
		},
	},
};

const store = new Vuex.Store({});

describe("TopBar", () => {
	let wrapper;

	beforeEach(() => {
		// Arrange
		wrapper = mount(TopBar, {
			localVue,
			store,
			mocks: {
				$route,
			},
		});
	});

	it("matches snapshot", () => {
		// Assert
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("renders properly", () => {
		// Assert
		const pageTitle = wrapper.findComponent(PageTitle);
		const iconButton = wrapper.findComponent(IconButton);
		const divs = wrapper.findAll("div");

		expect(pageTitle.exists()).toBe(true);
		expect(iconButton.exists()).toBe(true);
		expect(divs.exists()).toBe(true);
		expect(divs).toHaveLength(3);
	});

	[
		{ element: "div", clx: "top-bar-container" },
		{ element: "div", clx: "top-bar-inner-container" },
		{ element: "div", clx: "flex-center", cardinality: 2 },
		{ element: "div", clx: "flex-center-row", cardinality: 2 },
		{ element: "div", clx: "top-bar-actions-container" },
	].forEach(({ element, clx, cardinality = 1 }) =>
		it("has correct css classes", () => {
			// Act
			const elements = wrapper.findAll(element);
			const filteredElements = elements.filter((el) => el.classes(clx));

			// Assert
			expect(filteredElements).toHaveLength(cardinality);
		})
	);
});
