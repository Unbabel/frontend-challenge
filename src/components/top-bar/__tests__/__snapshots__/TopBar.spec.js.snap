// Vitest Snapshot v1

exports[`TopBar > matches snapshot 1`] = `
"<div data-v-83129ed2=\\"\\" class=\\"top-bar-container\\">
  <div data-v-83129ed2=\\"\\" class=\\"top-bar-inner-container flex-center flex-center-row\\"><span data-v-dde5e378=\\"\\" data-v-83129ed2=\\"\\" data-test=\\"page-title\\" class=\\"page-title\\"> Transcriptions </span>
    <div data-v-83129ed2=\\"\\" data-test=\\"top-bar-actions\\" class=\\"top-bar-actions-container flex-center flex-center-row\\"><button data-v-f509f990=\\"\\" data-v-83129ed2=\\"\\" data-test=\\"icon-button\\" aria-label=\\"Icon Button\\" class=\\"icon-button\\"><img data-v-f509f990=\\"\\" alt=\\"button-icon\\"></button><button data-v-f509f990=\\"\\" data-v-83129ed2=\\"\\" data-test=\\"icon-button\\" aria-label=\\"Icon Button\\" class=\\"icon-button\\"><img data-v-f509f990=\\"\\" alt=\\"button-icon\\"></button></div>
  </div>
</div>"
`;
